google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java

```diff
@@ -45,31 +70,200 @@
+    } catch (Exception e) {
+      if (isNotExceptionType(exceptionTypes, e)) throw e;
+    }
+    try {
+      instanceClient.deleteInstance(INSTANCE_NAME);
+    } catch (Exception e) {
+      if (isNotExceptionType(exceptionTypes, e)) throw e;
+    }
+    try {
+      imageClient.deleteImage(IMAGE_NAME);
+    } catch (Exception e) {
+      if (isNotExceptionType(exceptionTypes, e)) throw e;
+    }
+  }
+
+  private static boolean isNotExceptionType(
+      List<Class<? extends ApiException>> exceptionTypes, Exception e) {
+    for (Class<? extends ApiException> type : exceptionTypes) {
+      if (type.isInstance(e)) {
+        return false;
+      }
+    }
+    return true;
   }
 
   @AfterClass
   public static void tearDown() throws Exception {
-    diskTypeClient.close();
+    try {
+      removeResources(Arrays.asList(NotFoundException.class, PermissionDeniedException.class));
+    } finally {
+      diskTypeClient.close();
+      instanceClient.close();
+      subNetworkClient.close();
+      diskClient.close();
+      imageClient.close();
+    }
+  }
+
+  @Test
+  public void testDetachDisk() throws Exception {
+    long startAt = System.currentTimeMillis();
+    System.out.println("Clients created in " + (System.currentTimeMillis() - startAt) + "ms.");
+
+    // Fetch a subnetwork, assuming there is one. Projects start with a default network.
+    Iterable<SubnetworksScopedList> subnetworks =
+        subNetworkClient.aggregatedListSubnetworks(DEFAULT_PROJECT).iterateAll();
+    // TODO(andrealin): Implement futures for Compute methods.
+    Thread.sleep(10000);
+    Subnetwork subnetwork = subnetworks.iterator().next().getSubnetworksList().get(0);
+
+    NetworkInterface networkInterface =
+        NetworkInterface.newBuilder()
+            .setNetwork(subnetwork.getNetwork())
+            .setSubnetwork(
+                ProjectRegionSubnetworkName.of(DEFAULT_PROJECT, REGION, subnetwork.getName())
+                    .toString())
+            .build();
+    AttachedDisk bootDisk =
+        AttachedDisk.newBuilder()
+            .setAutoDelete(true)
+            .setBoot(true)
+            .setInitializeParams(
+                AttachedDiskInitializeParams.newBuilder()
+                    .setSourceImage(IMAGE_NAME.toString())
+                    .setDiskSizeGb(String.valueOf(100))
+                    .build())
+            .build();
+
+    // Insert an instance using the found subnetwork and its corresponding network (may not be the
+    // inserted network).
+    Instance instanceResource =
+        Instance.newBuilder()
+            .setName(INSTANCE_NAME.getInstance())
+            .addNetworkInterfaces(networkInterface)
+            .setMachineType(MACHINE_TYPE_NAME.toString())
+            .addDisks(bootDisk)
+            .build();
+    Operation op1 = instanceClient.insertInstance(PROJECT_ZONE_NAME, instanceResource);
+    System.out.println(String.format("Instance created: %s", op1.toString()));
```

Comment:
Does this still belong here?  If so, then perhaps a log statement would be better.